### PR TITLE
Add BranchMergeCommand for remote merging branches (with a few extra's)

### PR DESCRIPTION
**Pull request by sstok**
|Q            |A  |
|---          |---|
|Fixed Tickets|   |
|License      |MIT|
                   

This command does more the just merging, it also guards that the branches are merged according to
a configured workflow (eg no 2.3 to > 2.0).

The workflow is completely adjustable to ones needs, at the moment it supports semver and Git-flow.
And allows to configure constraints to which branches maybe to where.

Todo:

- [x] Testing
- [x] Fix missing commits-notes `--notes` when merging (GitHelper)

### CODE REVIEW

**Comment by sstok** [comment]
Ready for review @gushphp/contributor

File: src/Command/Branch/BranchMergeCommand.php [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+                    $sourceOrg.'/'.$sourceBranch,
+                    $targetOrg.'/'.$targetBranch
+                )
+            );
+
+            return self::COMMAND_SUCCESS;
+        } catch (CannotSquashMultipleAuthors $e) {
+            $this->getHelper('gush_style')->error(
+                [
+                    "Enable to squash commits when there are multiple authors.",
```
Comment (pierredup) [inline src/Command/Branch/BranchMergeCommand.php]: I think this should be `Unable to ...` ?

Comment (sstok) [inline src/Command/Branch/BranchMergeCommand.php]: That should also be fixed in the PullRequestMergeCommand :)

**Comment by cordoval** [comment]
please send those PRs as follow ups :+1:

**Comment by sstok** [comment]
Done, see #442

**Pull request merged**